File: test/cvdemo/Examples/Delphi/Via API/Unit1.pas
unit Unit1;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls, VirtualizerSDK;

type
  TForm1 = class(TForm)
    GroupBox1: TGroupBox;
    Button1: TButton;
    Button2: TButton;
    procedure Button1Click(Sender: TObject);
    procedure Button2Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.dfm}

procedure TForm1.Button1Click(Sender: TObject);

var
  Value, i: Integer;

begin
             
    Value := 0;

    // the following code, inside the Virtualizer macro will be converted
    // into virtual opcodes

    VIRTUALIZER_TIGER_WHITE_START;

	  for i := 0 to  10 do
      Value := Value * 2;

    VIRTUALIZER_TIGER_WHITE_END;

     MessageBox(0, 'The Virtualizer Macro #1 has successfully been executed', 'Virtualizer Macro', MB_OK + MB_ICONINFORMATION);

end;

procedure TForm1.Button2Click(Sender: TObject);

var
  Value, i: Integer;

begin

    Value := 0;

    // the following code, inside the ENCODE macro, will go encrypted
    // all the time and decrypted/encrypted again when it's executed

    VIRTUALIZER_TIGER_RED_START;

	  for i := 0 to  10 do
      Value := Value * 3;

    VIRTUALIZER_TIGER_RED_END;

     MessageBox(0, 'The Virtualizer Macro #1 has successfully been executed', 'Virtualizer Macro', MB_OK + MB_ICONINFORMATION);

end;

end.

File: test/cvdemo/Examples/C/CBuilder/Unit1.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Unit1.h"
#include "..\..\..\Include\C\VirtualizerSDK.h"

//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TForm1::Button1Click(TObject *Sender)
{
     int        value = 0;

     // the following code, inside the Virtualizer macro, will be
     // converted into Virtual opcodes

     VIRTUALIZER_TIGER_WHITE_START

     for (int i = 0; i < 10; i++)
     {
         value += value * i;
     }

     VIRTUALIZER_TIGER_WHITE_START

     MessageBox(NULL, "The Virtualizer Macro #1 has successfully been executed", "Code Virtualizer", MB_OK + MB_ICONINFORMATION);

}
//---------------------------------------------------------------------------

void __fastcall TForm1::Button2Click(TObject *Sender)
{
     int        value = 0;

     // the following code, inside the Virtualizer macro, will be
     // converted into Virtual opcodes

     VIRTUALIZER_TIGER_WHITE_START

     for (int i = 0; i < 10; i++)
     {
         value += value * i * 2;
     }

     VIRTUALIZER_TIGER_WHITE_START

     MessageBox(NULL, "The Virtualizer Macro #2 has successfully been executed", "Code Virtualizer", MB_OK + MB_ICONINFORMATION);

}
//---------------------------------------------------------------------------
